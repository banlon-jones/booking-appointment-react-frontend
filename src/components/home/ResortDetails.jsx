import React, { useState } from 'react';
import { useNavigate, useParams } from 'react-router-dom';
import Modal from 'react-bootstrap/Modal';
import { useForm } from 'react-hook-form';
import { useDispatch } from 'react-redux';
// eslint-disable-next-line no-unused-vars
import { createReservation } from '../../store/reservation/reservation';

const ResortDetails = () => {
  const { register, handleSubmit, formState: { errors } } = useForm();
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const [show, setShow] = useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  const { id } = useParams();
  const onSubmit = (data) => {
    dispatch(createReservation({ ...data, resort_id: id }));
    navigate('/reservations');
  };
  return (
    <div className="container">
      ResortDetails
      <span>{id}</span>
      <button type="button" className="btn btn-primary" onClick={handleShow}>
        Launch
      </button>
      <div>
        <Modal
          show={show}
          onHide={handleClose}
          backdrop="static"
          keyboard={false}
        >
          <Modal.Header closeButton>
            <Modal.Title>Make a Reservation</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <form onSubmit={handleSubmit(onSubmit)}>
              <div className="row">
                <div className="col-12">
                  <p> From </p>
                  {/* eslint-disable-next-line react/jsx-props-no-spreading */}
                  <input type="date" id="from" className="form-control" {...register('date_from', { required: true })} />
                  {errors.date_from && <span>This date is required</span>}
                </div>
                <div className="col-12">
                  <p> To </p>
                  {/* eslint-disable-next-line react/jsx-props-no-spreading */}
                  <input type="date" className="form-control" {...register('date_to', { required: true })} />
                  {errors.date_to && <span>This date is required</span>}
                </div>
              </div>
              <div className="m-3">
                <button type="submit" className="btn btn-success">
                  {' '}
                  submit
                </button>
              </div>
            </form>
          </Modal.Body>
        </Modal>
      </div>
    </div>
  );
};

export default ResortDetails;
