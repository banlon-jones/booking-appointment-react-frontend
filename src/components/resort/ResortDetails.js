/* eslint-disable comma-dangle */
/* eslint-disable no-unused-vars */

import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { BsChevronRight } from 'react-icons/bs';
import { useNavigate, useParams } from 'react-router-dom';
import { setRole } from '../../store/user/user';
import ReserveResort from './resort_components/ReserveResort';
import {
  selectResort,
  getResort,
} from '../../store/resort-details/resortDetail';
import resortClass from './ResortDetails.module.css';

function Resort() {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const { resortId } = useParams();
  useEffect(() => {
    dispatch(getResort(resortId));
    dispatch(setRole());
  }, []);

  const user = useSelector((state) => state.user);
  const resortDetails = useSelector(selectResort);
  return (
    <div className={resortClass.resort_container} data-testid="Resort-details">
      <div className={resortClass.inner_container}>
        <div className={resortClass.image_container}>
          <img src={resortDetails.image} alt="logo" />
        </div>

        <div className={resortClass.resort_details_container}>
          <h1>{resortDetails.name}</h1>
          <ul>
            <li>
              Country:
              <span>{resortDetails.country}</span>
            </li>
            <li>
              City:
              <span>{resortDetails.city}</span>
            </li>
            <li>
              Description:
              <p>{resortDetails.description}</p>
            </li>
            <li>
              Cost:
              <span>{`$${resortDetails.cost}`}</span>
            </li>
          </ul>
          <div className={resortClass.button_main_container}>
            <ReserveResort>
              Reserve
              <BsChevronRight className={resortClass.icon} />
            </ReserveResort>
            <button
              type="button"
              className={resortClass.button_container}
              style={{ display: user.role === 'admin' ? 'block' : 'none' }}
              onClick={() => {
                navigate(`/resorts/update/${resortDetails.id}`);
              }}
            >
              Update
              <BsChevronRight className={resortClass.icon} />
            </button>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Resort;
