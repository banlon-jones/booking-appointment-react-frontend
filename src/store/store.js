/* eslint-disable import/no-named-as-default */
/* eslint-disable import/no-named-as-default-member */
import { configureStore } from '@reduxjs/toolkit';
import userSlice from './user/user';
import resortDetailSlice from './resort-details/resortDetail';

const store = configureStore({
  reducer: {
    user: userSlice,
    resortDetail: resortDetailSlice,
  },
});

export default store;
