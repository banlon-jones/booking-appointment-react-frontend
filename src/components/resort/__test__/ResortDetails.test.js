/* eslint-disable comma-dangle */
import React from 'react';
import { cleanup, screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../../store/store';
import ResortDetails from '../ResortDetails';

afterEach(cleanup);

describe('React Test for AddResort', () => {
  test('Add RssortTest', () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <ResortDetails />
        </Provider>
      </BrowserRouter>
    );
    const ResortElement = screen.getByTestId('Resort-details');
    expect(ResortElement).toBeInTheDocument('City');
  });
});

describe('Add Resort Snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <ResortDetails />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
