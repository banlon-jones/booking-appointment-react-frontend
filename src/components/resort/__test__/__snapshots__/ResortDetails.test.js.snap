// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Resort Snapshot renders correctly 1`] = `
<div
  className="resort_container"
  data-testid="Resort-details"
>
  <div
    className="inner_container"
  >
    <div
      className="image_container"
    >
      <img
        alt="logo"
      />
    </div>
    <div
      className="resort_details_container"
    >
      <h1 />
      <ul>
        <li>
          Country:
          <span />
        </li>
        <li>
          City:
          <span />
        </li>
        <li>
          Description:
          <p />
        </li>
        <li>
          Cost:
          <span>
            $undefined
          </span>
        </li>
      </ul>
      <div
        className="button_main_container"
      >
        <button
          className="button_container"
          data-testid="reserve-button"
          onClick={[Function]}
          type="button"
        >
          Reserve
        </button>
        <div />
        <button
          className="button_container"
          onClick={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="button"
        >
          Update
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
              fillRule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
